browser/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -0,0 +1,499 @@
+    for (size_t i = 0; i < cf_exceptions_list->GetSize(); i++) {
+      exceptions_.push_back(cf_exceptions_list->GetList()[i].GetString());
+    }
+  }
+  base::ListValue* hide_selectors_list;
+  base::ListValue* force_hide_selectors_list = nullptr;
+  if (resources_dict->GetList("hide_selectors", &hide_selectors_list)) {
+    if (enabled_1st_party_cf_filtering_) {
+      force_hide_selectors_list = hide_selectors_list;
+    } else {
+      std::string cosmeticFilterConsiderNewSelectors_script =
+          "(function() {"
+            "let nextIndex = window.cosmeticStyleSheet.rules.length;";
+      std::string json_selectors;
+      base::JSONWriter::Write(*hide_selectors_list, &json_selectors);
+      if (!json_selectors.empty()) {
+        cosmeticFilterConsiderNewSelectors_script += "const selectors = " +
+            json_selectors;
+      } else {
+        cosmeticFilterConsiderNewSelectors_script += "const selectors = []";
+      }
+      cosmeticFilterConsiderNewSelectors_script += ";"
+          "selectors.forEach(selector => {"
+            "if (!window.allSelectorsToRules.has(selector)) {"
+              "let rule = selector + '{display:none !important;}';"
+              "window.cosmeticStyleSheet.insertRule(`${rule}`, nextIndex);"
+              "window.allSelectorsToRules.set(selector, nextIndex);"
+              "nextIndex++;"
+              "window.firstRunQueue.add(selector);"
+            "}"
+          "});"
+          "if (!document.adoptedStyleSheets.includes("
+                "window.cosmeticStyleSheet)){"
+              "document.adoptedStyleSheets = [window.cosmeticStyleSheet];"
+          "};"
+          "})();";
+      if (hide_selectors_list->GetSize() != 0) {
+        auto* frame_host = content::RenderFrameHost::FromID(frame_id);
+        if (!frame_host)
+          return;
+        frame_host->ExecuteJavaScriptInIsolatedWorld(
+            base::UTF8ToUTF16(cosmeticFilterConsiderNewSelectors_script),
+            base::NullCallback(), ISOLATED_WORLD_ID_CHROME_INTERNAL);
+      }
+    }
+  }
+
+  std::string styled_stylesheet = "";
+  if (force_hide_selectors_list!= nullptr &&
+      force_hide_selectors_list->GetSize() != 0) {
+    for (size_t i = 0; i < force_hide_selectors_list->GetSize(); i++) {
+      if (i != 0) {
+        styled_stylesheet += ",";
+      }
+      styled_stylesheet += force_hide_selectors_list->GetList()[i].GetString();
+    }
+    styled_stylesheet += "{display:none!important;}\n";
+  }
+  base::DictionaryValue* style_selectors_dictionary = nullptr;
+  if (resources_dict->GetDictionary("style_selectors",
+      &style_selectors_dictionary)) {
+    for (const auto& it : style_selectors_dictionary->DictItems()) {
+      for (size_t i = 0; i < it.second.GetList().size(); i++) {
+        if (i != 0) {
+          styled_stylesheet += ";";
+        } else {
+          styled_stylesheet += it.first + "{";
+        }
+        styled_stylesheet += it.second.GetList()[i].GetString();
```

Comment:
i think we also need to do validation here since it gets injected as JS in `cosmeticFilterConsiderNewSelectors_script`. it would be safer if this were written to a CSS file that got included in the HTML so at least there's no chance of anything in the resource being parsed as javascript.

@antonok-edm @pes10k thoughts?

Reply:
I don't think we can include css file in already loaded page, I will check that to use json kind object as well

Reply:
What about?
1. Serialize the CSS to inject as JSON or some inert value
2. Inject the JSON into the page
3. Inject prewritten script into the page that iterates over the values in the JSON, and uses the existing "create and inject a new CSS Page into the document" methods? Something like https://github.com/brave/brave-core/blob/af9bb0722c0953ccb4114fca4dbff8138cc1126b/components/brave_extension/extension/brave_extension/content_cosmetic.ts#L580 that will throw if its not valid CSS?

Reply:
addressed